include/sir/platform.h

```diff
@@ -84,7 +84,24 @@
 #  endif
 #  if defined(__IMPORTC__)
 #   undef __HAVE_ATOMIC_H__
+#   undef __SIZEOF_INT128__
 #   define _BITS_FLOATN_H 1
+#   undef __builtin_bswap16
+#   define __builtin_bswap16
+#   undef __builtin_bswap32
+#   define __builtin_bswap32
+#   undef __builtin_bswap64
```

Comment:
I think it's about time we split `platform.h` up just like we did with `Makefile`... it's getting unruly.

I would like to move all these platform-specific preprocessor blocks into their own `name-platform.h` that is `#include`d in the main header.

Reply:
It's not too bad right now, at least all the ugliness is kept in one place.  And since the ordering matters, having to split things up wouldn't necessarily make it more readable, I don't think.  There are five different sections that switch on _AIX, for example.

Reply:
Sooner or later (probably sooner rather than later), it's gonna get out of hand... if necessary we can have `name-platform-n.h` and include them in the appropriate places. Readability isn't my concern–it's maintainability of that gnarly header file with 8 million preprocessor switches

Reply:
We can merge this for now, but I'm not super stoked on adding like 100 lines of preproc for a compiler nobody I know has or ever will use :P